test/numbers.jl

```diff
@@ -460,6 +460,8 @@ end
     @test sign(-0//1) == 0
     @test sign(1//0) == 1
     @test sign(-1//0) == -1
+    @test sign(pi) == 1
+    @test sign(pi) === -sign(-pi)
```

Comment:
Note that `-pi` is a `Float64`, so your second new test doesn't really test anything...

Reply:
I believe, in addition to `@test sign(-pi) == -1`, it tests `sign()` returns the same type. This may be also beneficial for other cases for detecting possible type instability.

Reply:
@fingolfin we reasoned that since `-pi` goes to Float64, `sing(pi)` should also return a Float64. This is to test that